Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Dashboard/FRFSTicket.hs

```diff
@@ -16,38 +16,281 @@ where
+
+  unprocessedFares <- forM fareUpdates $ \row -> do
+    let amountVal = row.amount
+    case highPrecMoneyFromText amountVal of
+      Nothing -> do
+        let message = "Invalid amount format for route " <> row.routeCode <> " (" <> row.startStopCode <> " -> " <> row.endStopCode <> ")"
+        throwError (InvalidRequest message)
+      Just value -> do
+        existingFares <- QRSF.findByRouteStartAndStopCode row.routeCode row.startStopCode row.endStopCode
+        case existingFares of
+          [] -> do
+            logInfo $ "No matching fare found for route " <> row.routeCode <> " with startStopCode " <> row.startStopCode <> " and endStopCode " <> row.endStopCode
+            pure [(row.routeCode, row.startStopCode, row.endStopCode)]
+          _ -> do
+            QRSF.updateFareByRouteCodeAndStopCodes value row.routeCode row.startStopCode row.endStopCode
+            logInfo $ "Updated fare for route " <> row.routeCode <> " from " <> row.startStopCode <> " to " <> row.endStopCode <> " with amount " <> show value
+            pure []
+
+  let allUnprocessedFares = concat unprocessedFares
+
+  let totalUpdates = length fareUpdates
+  let successfulUpdates = totalUpdates - length allUnprocessedFares
+
+  if successfulUpdates == totalUpdates
+    then do
+      pure $
+        API.Types.RiderPlatform.Management.FRFSTicket.UpsertRouteFareResp
+          { unprocessedRouteFares = [],
+            success = "All fields updated successfully"
+          }
+    else do
+      let unprocessedRouteFares =
+            map
+              ( \(routeCode, startStopCode, endStopCode) ->
+                  "Route: " <> routeCode <> ", Start Stop: " <> startStopCode <> ", End Stop: " <> endStopCode
+              )
+              allUnprocessedFares
+      pure $
+        API.Types.RiderPlatform.Management.FRFSTicket.UpsertRouteFareResp
+          { unprocessedRouteFares = unprocessedRouteFares,
+            success = "Partial update completed"
+          }
+  where
+    readCsv :: FilePath -> Environment.Flow [FareUpdateCSVRow]
+    readCsv csvFile = do
+      csvData <- L.runIO $ BS.readFile csvFile
+      case (decodeByName $ LBS.fromStrict csvData :: Either String (Header, V.Vector FareUpdateCSVRow)) of
+        Left err -> throwError (InvalidRequest $ show err)
+        Right (_, v) -> return $ V.toList v
 
 getFRFSTicketFrfsRouteStations :: (Kernel.Types.Id.ShortId Domain.Types.Merchant.Merchant -> Kernel.Types.Beckn.Context.City -> Kernel.Prelude.Maybe (Data.Text.Text) -> Kernel.Prelude.Int -> Kernel.Prelude.Int -> BecknV2.FRFS.Enums.VehicleCategory -> Environment.Flow [API.Types.RiderPlatform.Management.FRFSTicket.FRFSStationAPI])
-getFRFSTicketFrfsRouteStations _merchantShortId _opCity searchStr limit offset vehicleType = do error "Logic yet to be decided" searchStr limit offset vehicleType
+getFRFSTicketFrfsRouteStations merchantShortId opCity searchStr limit offset vehicleType = do
+  let limitVal = toInteger $ fromMaybe 10 (Just limit)
```

Comment:
Why fromMaybe is required ? It's already a mandatory field on UI.
You may remove fromMaybe.